ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Genesis/Governor.hs

```diff
@@ -227,21 +227,13 @@ densityDisconnect (GenesisWindow sgen) (SecurityParam k) states candidateSuffixe
                max (AF.headSlot candidateSuffix) latestSlot
             >= NotOrigin firstSlotAfterGenesisWindow
 
-          -- If the peer is idling, we assume it has no more headers to send.
-          --
           -- If the slot of the latest header we know of is _after_ the end of
           -- the Genesis window (either because the candidate fragment extends
           -- beyond it or because we are waiting to validate a header beyond the
           -- forecast horizon that we already received), there can be no headers
           -- in between and 'potentialSlots' is 0.
-          --
-          -- If the peer has more headers that it hasn't sent yet, each slot
-          -- between the latest header we know of and the end of the Genesis
-          -- window could contain a block, so the upper bound for the total
-          -- number of blocks in the window is the sum of the known blocks and
-          -- that number of remaining slots.
           potentialSlots =
-            if idling || hasBlockAfter then 0
+            if hasBlockAfter then 0
             else unknownTrailingSlots
```

Comment:
What is the motivation for this change?

Reply:
I tried to explain it in [this comment](https://github.com/IntersectMBO/ouroboros-consensus/pull/1068#issuecomment-2066650093).

In essence, it gives GDD the capacity to query if all headers have been received in the genesis window after the intersection. Before this change, we could only ask if a header has been received after the genesis window.

Reply:
> Eventually, the LoP wil guarantee that they will either send us a header beyond the Genesis window or send MsgAwaitReply.

I think so.

> such that changes to these notions are reflected in updated documentation.

The upper bound is only used in `densityDisconnect` at the moment.

I guess whatever documentation exists will need to be updated to explain the nuances here indeed.